include/EASTL/sort.h

```diff
@@ -205,77 +205,144 @@ namespace eastl
-				// but std STL iterators may have a problem with it. Dinkumware STL, for example, will assert.
-				// To do: Fix this loop to not decrement like so.
-				for(--iCurrent; (iNext != first) && compare(temp, *iCurrent); --iNext, --iCurrent)
-				{
-					EASTL_VALIDATE_COMPARE(!compare(*iCurrent, temp)); // Validate that the compare function is sane.
-					*iNext = *iCurrent;
-				}
-
-				*iNext = temp;
-			}
-		}
-	} // insertion_sort
-
-
-
-	template <typename BidirectionalIterator>
-	void insertion_sort(BidirectionalIterator first, BidirectionalIterator last)
+	    typedef typename eastl::iterator_traits<Iter>::value_type value_type;
+	
+	    if (base!=past)
+	    {   //this could perhaps be written more idiomatic: for (T *left=base, *sorted=base+1; sorted!=past; left=sorted++)
+	        for (Iter left=base, sorted=base; ++sorted!=past; left=sorted)
+	        {
+	            if (cmp(*sorted, *left))
+	            {
+	                const value_type val=*sorted;//should we care about doing moves instead?
+	                Iter right=sorted;
+	
+	                do
+	                {
+	                    EASTL_VALIDATE_COMPARE(!cmp(*left, val));//validate cmp is sane by checking other way
+	                    *right=*left;
+	                }while ((right=left)!=base && cmp(val, *--left));
+	
+	                *right=val;
+	            }
+	        }//outer loop
+	    }//non-empty range
+	}//insertion_sort (Compare)
+	
+	//default less-than version
+	template<class Iter>//Bidirectional Iterator, though something like a list shouldn't use this
+	void insertion_sort(Iter const base, Iter const past)
 	{
-		typedef typename eastl::iterator_traits<BidirectionalIterator>::value_type value_type;
-
-		if(first != last)
-		{
-			BidirectionalIterator iCurrent, iNext, iSorted = first;
-
-			for(++iSorted; iSorted != last; ++iSorted)
-			{
-				const value_type temp(*iSorted);
-
-				iNext = iCurrent = iSorted;
-
-				// Note: The following loop has a problem: it can decrement iCurrent to before 'first'.
-				// It doesn't dereference the iterator, but std STL disallows that operation. This isn't 
-				// a problem for EASTL containers and ranges, as they support a single decrement of first,
-				// but std STL iterators may have a problem with it. Dinkumware STL, for example, will assert.
-				// To do: Fix this loop to not decrement like so.
-				for(--iCurrent; (iNext != first) && (temp < *iCurrent); --iNext, --iCurrent)
-				{
-					EASTL_VALIDATE_COMPARE(!(*iCurrent < temp)); // Validate that the compare function is sane.
-					*iNext = *iCurrent;
-				}
-
-				*iNext = temp;
-			}
-		}
-	} // insertion_sort
-
+	    typedef typename eastl::iterator_traits<Iter>::value_type value_type;
+	
+	    if (base!=past)
+	    {   //this could perhaps be written more idiomatic: for (T *left=base, *sorted=base+1; sorted!=past; left=sorted++)
+	        for (Iter left=base, sorted=base; ++sorted!=past; left=sorted)
```

Comment:
The format and variable names aren't in the EASTL standard.  Look at or use the .clang-format file in the repo for what to do there.